### DISCUSSION TITLE
Is there a way to set user with password not in cleartext?

### DISCUSSION THREAD

**Discussion by kuolemaaa** [nanomq/nanomq#1933] [Q&A] [answered]
Reading the docs, I find very weird to keep a file with username and password in cleartext as shown in this page https://nanomq.io/docs/en/latest/access-control/login.html

Isnt there a way to compute and save some sort of hash of the password like mosquitto does ( `mosquitto_passwd` )?

**Comment by JaylinYu** [comment]
Already done so In https://github.com/nanomq/nanomq/pull/1877

**Reply by kri164** [reply]
Can you explain, how to use it, please?
Current version of nanomq_cli doesnt support encryption subcommand.

**Reply by JaylinYu** [reply]
That PR was obsoleted....
We will find a more appropriate way to achieve this. But not in near future, I am super busy so any help is appreciated

**Reply by JaylinYu** [reply]
Now it is supported in EE ver... if this concerns you .... 
Sorry, I will try to move it to OS later

**Comment by JaylinYu** [comment, marked as answer]
Now the encrypted password feature is embedded with EE ver... if this concerns you ....
Sorry, I will try to move it to OS later
